### PR TITLE
Fix test_raylet_pending_tasks test case failed

### PR DESCRIPTION
**Pull request by ffbin**
## Why are these changes needed?
Test case of test_raylet_pending_tasks will create ParentActor first, then ParentActor will create ActorRequiringGPU actor. Now reference counter will terminate ParentActor because we do not get ParentActor actor handler in the driver code.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Comment by wumuzi520** [comment]
LGTM!

File: python/ray/tests/test_metrics.py [hunk 1]
```diff
@@ -300,7 +300,7 @@ class ParentActor:
         def __init__(self):
             self.a = [ActorRequiringGPU.remote() for i in range(4)]
 
-    ParentActor.remote()
+    parent_actor = ParentActor.remote()
```
Comment (wumuzi520) [inline python/ray/tests/test_metrics.py]: parent_actor_handle = ParentActor.remote()

**Comment by raulchen** [comment]
@stephanie-wang , is there an option to control whether an actor should be killed when all handles are gone? I think we need such an option. Because not all actors need to be killed. For example, a streaming source actor just keeps reading data from somewhere, no one will call it.

**Comment by kfstorm** [comment]
FYI, the test case failure was triggered by [this change](https://github.com/ray-project/ray/pull/7346/files#diff-db1b8e62933183879f43554ada5ec439L105-L108). Previously if only the kill request was initiated but no tasks were submitted to the actor, the kill request would actually stay in the `pending_force_kills_` because `pending_requests_.count(actor_id)` is 0.

**Comment by stephanie-wang** [comment]
> @stephanie-wang , is there an option to control whether an actor should be killed when all handles are gone? I think we need such an option. Because not all actors need to be killed. For example, a streaming source actor just keeps reading data from somewhere, no one will call it.

Detached actors are not killed when there are no more handles.

File: python/ray/tests/test_metrics.py [hunk 2]
```diff
@@ -300,7 +300,8 @@ class ParentActor:
...
+    assert parent_actor is not None
```
Comment (stephanie-wang) [inline python/ray/tests/test_metrics.py]: I don't think you need this line.

**Review by stephanie-wang — APPROVED** [review]
Thanks!

File: python/ray/tests/test_metrics.py (hunk 2 above)
Comment (ffbin) [reply python/ray/tests/test_metrics.py]: This line is to solve the lint error: local variable 'parent' actor 'is assigned to but never used

**Pull request merged**